stream-chat-android-core/src/main/java/io/getstream/chat/android/client/utils/Result.kt

```diff
@@ -17,221 +17,129 @@
 package io.getstream.chat.android.client.utils
 
 import io.getstream.chat.android.client.errors.ChatError
-import io.getstream.chat.android.core.internal.InternalStreamChatApi
 
 /**
- *  A class which encapsulates a successful outcome with a value of type [T] or a failure with [ChatError].
+ *  A class which encapsulates a successful outcome with a value of type [A] or a failure with [ChatError].
  */
-public class Result<T : Any> private constructor(
-    private val data: T?,
-    private val error: ChatError?,
-) {
-
-    @Suppress("DEPRECATION")
-    public constructor(data: T) : this(data, null)
-
-    @Suppress("DEPRECATION")
-    public constructor(error: ChatError) : this(null, error)
+public sealed class Result<out A : Any> {
 
     /**
-     * Returns true if a request of payload response has been successful.
```

Comment:
What do you think adding some extensions for Result class below?

```kotlin
public val Result<*>.isSuccess: Boolean
    inline get() = this is Result.Success

public val Result<*>.isFailure: Boolean
    inline get() = this is Result.Failure
```

There're many useful cases for this and we also don't need to change Java codes to check `instanceOf`.

Reply:
Good idea! Let me update the code